mdnsd/packet.c

```diff
@@ -957,11 +964,83 @@ pkt_parse_rr(u_int8_t **pbuf, u_int16_t *len, struct rr *rr)
 	*len  -= n;
 	/* Make sure rr packet len is ok */
 	if (*len < 8) {
-		log_debug("Unexpected packet len");
+		log_warnx("Unexpected packet len");
 		return (-1);
 	}
 	GETSHORT(rr->rrs.type, *pbuf);
 	*len -= INT16SZ;
+
+	/*
+	 * The T_OPT type is handled differently from other RR types.
+	 * See RFC 2671 for details.
+	 */
+
```

Comment:
This reads good to me but I didn't have time to check the RFC yet.

It would be a bit more clean to do something like:

if (rr->rrs.type == T_OPT)
   goto switch;

then on the 
case T_OPT you do this whole block and return.
This way you can escape the stuff you don't want, and code still looks like a switch